cdap-support-bundle/src/main/java/io/cdap/cdap/support/services/SupportBundleGenerator.java

```diff
@@ -218,4 +307,49 @@ private NamespaceId validNamespace(NamespaceId namespace) throws Exception {
 
     return namespace;
   }
+
+  private SupportBundlePipelineStatus collectSupportBundleTaskStatus(
+    Set<SupportBundleTaskStatus> supportBundleTaskStatusSet) {
+    CollectionState systemLogTaskStatus = CollectionState.INVALID;
+    CollectionState pipelineInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeLogTaskStatus = CollectionState.INVALID;
+    CollectionState vmInfoTaskStatus = CollectionState.INVALID;
+    Map<SupportBundleTaskType, CollectionState> taskStatusMap = new HashMap<>();
+    for (SupportBundleTaskStatus supportBundleTaskStatus : supportBundleTaskStatusSet) {
+      SupportBundleTaskType currentTaskType = SupportBundleTaskType.valueOf(supportBundleTaskStatus.getType());
+      taskStatusMap.put(currentTaskType, supportBundleTaskStatus.getStatus());
+      if (supportBundleTaskStatus.getSubTasks().size() > 0) {
+        for (SupportBundleTaskStatus subTaskSupportBundleTaskStatus : supportBundleTaskStatus.getSubTasks()) {
+          SupportBundleTaskType currentSubTaskType =
+            SupportBundleTaskType.valueOf(subTaskSupportBundleTaskStatus.getType());
+          taskStatusMap.put(currentSubTaskType, subTaskSupportBundleTaskStatus.getStatus());
```

Comment:
What is the purpose of adding these subtasks to the map? It seems like it's not added to the `SupportBundlePipelineStatus` constructor.

Reply:
yeah we introduce SupportBundleRuntimeInfoTask and SupportBundleRuntimeLogTask as two sub tasks of SupportBundlePipelineInfoTask because they share the same namespace, application details, so all these task status is inside SupportBundlePipelineStatus constructor